plugins/powerd/fu-plugin-powerd.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2021 Twain Byrnes <binarynewts@google.com>
+ * Copyright (C) 2021 George Popoola <gpopoola@google.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#define MINIMUM_BATTERY_PERCENTAGE_FALLBACK 10
+
+struct FuPluginData {
+	GDBusProxy *proxy; /* nullable */
+};
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	if (data->proxy != NULL)
+		g_object_unref (data->proxy);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	g_autoptr(GDBusConnection) connection = NULL;
+	FuPluginData *data = fu_plugin_get_data (plugin);
+
+	/* Establishing DBus connection and proxy for method call to powerd */
+	connection = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, error);
+
+	if (connection == NULL) {
+		g_prefix_error (error,
+				"Connection with system was not established: ");
+		return FALSE;
+	}
+
+	data->proxy = g_dbus_proxy_new_sync (connection,
+					     G_DBUS_PROXY_FLAGS_NONE,
+					     NULL,
+					     "org.chromium.PowerManager",
+					     "/org/chromium/PowerManager",
+					     "org.chromium.PowerManager",
+					     NULL,
+					     error);
+
+	if (data->proxy == NULL) {
+		g_prefix_error (
+		    error,
+		    "Failed to establish proxy for power manager server: ");
+		return FALSE;
+	}
+
+	name_owner = g_dbus_proxy_get_name_owner (data->proxy);
```

Comment:
You'll need to declare name_owner.

From CI:
```
../dist-unpack/fwupd-1.6.2/plugins/powerd/fu-plugin-powerd.c: In function ‘fu_plugin_startup’:
../dist-unpack/fwupd-1.6.2/plugins/powerd/fu-plugin-powerd.c:64:9: error: ‘name_owner’ undeclared (first use in this function)
   64 |         name_owner = g_dbus_proxy_get_name_owner (data->proxy);
      |         ^~~~~~~~~~
../dist-unpack/fwupd-1.6.2/plugins/powerd/fu-plugin-powerd.c:64:9: note: each undeclared identifier is reported only once for each function it appears in
```
Put this at the end of the declarations for _startup.
`	g_autofree gchar *name_owner = NULL;`